File: elixir/intersection_of_two_arrays.ex
defmodule Solution do
  @spec intersect(nums1 :: [integer], nums2 :: [integer]) :: [integer]
  def intersect(nums1, nums2) do
    f1 = Enum.frequencies(nums1)
    f2 = Enum.frequencies(nums2)

    Enum.reduce(f1, [], fn {n, c}, acc -> List.duplicate(n, min(c, Map.get(f2, n, 0))) ++ acc end)
  end
end

File: go/intersection_of_two_arrays.go
package main

func intersection(nums1 []int, nums2 []int) []int {
	seen := map[int]int{}
	for _, v := range nums1 {
		seen[v] = 1
	}
	for _, v := range nums2 {
		if seen[v] == 1 {
			seen[v] += 1
		}
	}
	r := []int{}
	for i, v := range seen {
		if v == 2 {
			r = append(r, i)
		}
	}
	return r
}
